etl/planning_data/obtain_livestream_data_and_load_into_database.py

```diff
@@ -0,0 +1,310 @@
+                "uprn": uprn,
+                "status": status,
+                "status_before_aliasing": status_before_aliasing,
+                "status_explanation_note": status_explanation_note,
+                "data_source": "Greater London Authority's Planning London DataHub",
+                "data_source_link": None,
+                "address": address_data.planning_data_entry_to_address(entry),
+            }
+            if entry["address"] != None:
+                maximum_address_length = 300
+                if len(entry["address"]) > maximum_address_length:
+                    print("address is too long, shortening", entry["address"])
+                    entry["address"] = entry["address"][0:maximum_address_length]
+            if date_in_future(entry["registered_with_local_authority_date"]):
+                print("registered_with_local_authority_date is treated as invalid:", entry["registered_with_local_authority_date"])
+                # Brent-87_0946 has "valid_date": "23/04/9187"
+                entry["registered_with_local_authority_date"] = None
+
+            if date_in_future(entry["decision_date"]):
+                print("decision_date is treated as invalid:", entry["decision_date"])
+                entry["decision_date"] = None
+
+            if date_in_future(entry["last_synced_date"]):
+                print("last_synced_date is treated as invalid:", entry["last_synced_date"])
+                entry["last_synced_date"] = None
+
+            if "Hackney" in entry["application_id"]:
+                if entry["application_url"] != None:
+                    if "https://" not in entry["application_url"]:
+                        entry["application_url"] = "https://developmentandhousing.hackney.gov.uk" + entry["application_url"]
+            insert_entry(cursor, entry)
+        except TypeError as e:
+            print()
+            print()
+            print()
+            print(e)
+            print()
+            show_dictionary(entry)
+            raise e
+
+
+def date_in_future(date):
+    if date == None:
+        return False
+    return date > datetime.datetime.now()
+
+
+def query(search_after):
+    headers = {
+        'X-API-AllowRequest': os.environ['PLANNNING_DATA_API_ALLOW_REQUEST_CODE'],
+        # Already added when you pass json= but not when you pass data=
+        # 'Content-Type': 'application/json',
+    }
+    json_data = {
+        'size': 10000,
+        'sort': [
+            {
+                'last_updated': {
+                    'order': 'desc',
+                    'unmapped_type': 'boolean',
+                },
+            },
+        ],
+        'stored_fields': [
+            '*',
+        ],
+        '_source': {
+            'excludes': [],
+        },
+        'query': {
+            'bool': {
+                'must': [
+                    {
+                        'range': {
+                            'valid_date': {
+                                'gte': '01/01/1021',
```

Comment:
1021 is a long time ago - was that meant to be 2021?

Reply:
In this case it was intended to cover all data (as one point I tested it and it needed some minimum date).

There is scattering of some old data like something from 1968, so in theory they may add older ones. Not sure when the first equivalent of planning permission appeared in London, so I set date that would catch everything.

Reply:
Though I will add comment explaining situation and retry is this weird filter avoidable.